pkg/hintrunner/zero/zerohint_memcpy.go

```diff
@@ -3,9 +3,54 @@ package zero
 import (
 	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/core"
 	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
+	"github.com/NethermindEth/cairo-vm-go/pkg/utils"
+	"github.com/NethermindEth/cairo-vm-go/pkg/vm/memory"
+
 	VM "github.com/NethermindEth/cairo-vm-go/pkg/vm"
 )
 
+func newMemcpyContinueCopyingHint(output hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "MemcpyContinueCopying",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> n -= 1
+			//> ids.continue_copying = 1 if n > 0 else 0
+
+			//> n-=1
+			n, err := ctx.ScopeManager.GetVariableValue("n")
+			if err != nil {
+				return err
+			}
+
+			n = n.(uint64) - 1
```

Comment:
Are we sure it should be `uint64` and not just a `felt`? I quickly went through some code in the cairo codebase and `uint64` most likely is fine. But the code does seem to indicate felt.

eg.
```
func append_felts{data: felt*}(len: felt, arr: felt*) {
    memcpy(dst=data, src=arr, len=len);
    let data = data + len;
    return ();
}
```
Here the `len` variable is of type `felt`. So maybe safer to just use that? cc @cicr99

Reply:
Type casting like `n = n.(uint64) - 1` can also fail.

I would check and return an error in that case.

eg:

```
n, ok = n.(uint64) - 1
if !ok {
    return fmt.Errorf("value: %s is not a uint64", n)
}

Reply:
so we should work with `n` as uint or change it to a felt? in case to treat `n` as felt, how can I cast the type to felt?

Reply:
Probably better with felt, @cicr99  ?

Reply:
So the thing here is the variable was assigned in the scope with a felt value. I think at the moment of definition you have `n = ids.len`, where `len` is a felt, so when you access the variable, typecasting it to uint64 should not work as it would be an `Element` (according to the library we are using for felts). Then I would go with felt here.

This being said, one note on this. In practice, the actual value of n cannot be a value of 252 bits since that wouldn't be feasible to compute, as this is representing the length of an array. n should fit in uint64. So if from the moment this variable is assigned to the scope we do the conversion, then your code would be good as it is. This will imply modifying a different part of the code. Plus, I think this is a situation that has happened other times in the code, so I would like to handle that in a different PR. I'm creating an issue for this as an optimization task

Reply:
Issue #383

Reply:
@cicr99 gotcha, so taking that into consideration I think this PR is ready, right?

Reply:
@danielcdz atm it will be best to work with felt here, as the variable you are using is saved as a felt in a previous hint. You'll need to modify your code for that. For felts we use the `Element` type from https://github.com/Consensys/gnark-crypto/tree/master/ecc/stark-curve/fp

Reply:
@cicr99 using felt now!